src/scripts/ci/publish.ts

```diff
@@ -16,121 +17,151 @@
  * @see {@link https://www.notion.so/arrivalms/Versioning-5bf1876eb2f142339d719f818cc2250c}
  */
 
+import * as path from 'path';
 import * as fs from 'fs';
 import * as childProcess from 'child_process';
 import { Dictionary } from '@eigenspace/common-types/src/types/dictionary';
 import { ArgumentParser } from '@eigenspace/argument-parser';
 
 const exists = require('npm-exists');
-
-const currentDir = process.cwd();
-const packageJson = require(`${currentDir}/package.json`);
 const exec = childProcess.execSync;
 
 const argParser = new ArgumentParser();
 const argv = argParser.get(process.argv.slice(2));
 
-// Get dependency suffix (branch name)
-const { name, version } = packageJson;
-const distParam = argv.get('dist') as string | undefined;
-
-const dist = distParam || fs.existsSync('./dist') && './dist' || currentDir;
+const projectPaths: string[] = argv.get('projectPaths') as string[] || ['/'];
 
 let currentBranch = getCurrentBranchName();
 
 if (!Boolean(currentBranch)) {
-    // eslint-disable-next-line no-console
     console.log('Current branch is undefined!');
     process.exit(1);
 }
 
-// eslint-disable-next-line no-console,no-console
 console.log('branch:', currentBranch);
-// eslint-disable-next-line no-console
-console.log('package to publish:', dist);
-
-// We consider snapshot as any non-master dependency version.
-// Every snapshot dependency version should be prerelease version that has suffix contains branch name
-const isSnapshotVersion = currentBranch !== 'master';
-const suffix = prepareSuffix(isSnapshotVersion ? `-${currentBranch}` : '');
-const fullVersion = `${name}@${version}${suffix}`;
-
-// If branch name is not master, i.e. there is no suffix
-if (isSnapshotVersion) {
-    // eslint-disable-next-line no-console
-    console.log('start publishing snapshot package...');
-    // Then remove dependency from registry
-    run(`npm unpublish ${fullVersion}`);
-    publish(`${version}${suffix}`);
-
-    if (dist === currentDir) {
-        // Remove snapshot version from package.json
-        run('git checkout package.json');
-    }
 
-} else {
-    // eslint-disable-next-line no-console
-    console.log('start publishing release package...');
-    exists(name).then((projectExists: Boolean) => {
+const projectPathsThatShouldBeIncrementedInDevBranch: string[] = [];
+projectPaths.forEach(async projectPath => await incrementPackagesInTargetBranch(projectPath));
+
+const isAllProjectsHaveSnapshotBranch = !projectPathsThatShouldBeIncrementedInDevBranch.length;
```

Comment:
Doesn’t it turn out that if we are in the master branch, then all internal projects need to be updated?